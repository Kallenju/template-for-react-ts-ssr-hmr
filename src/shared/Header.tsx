import React from 'react';
import styles from './header.module.styl';

export default function Header(): React.ReactElement {
  return (
    <header>
      <h1 className={styles.example}>Hello React</h1>
    </header>
  );
}
