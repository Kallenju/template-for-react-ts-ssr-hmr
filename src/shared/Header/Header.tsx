import React, { useState } from 'react';
import styles from './header.module.styl';

export function Header(): React.ReactElement {
  const [count, setCount] = useState(0);
  function handleClick() {
    setCount(count + 1);
  }
  return (
    <header>
      <h1 className={styles.example}>Hello React</h1>
      <button className={styles.example} onClick={handleClick}>
        Clicked {count} times
      </button>
      <picture>
        <source
          type="image/webp"
          width="637"
          height="118"
          srcSet="/static/images/raster/test-image/test-image.webp"
        />
        <img
          className={styles['preview__image']}
          src="/static/images/raster/test-image/test-image.jpg"
          alt="Test image"
          width="637"
          height="118"
        />
      </picture>
    </header>
  );
}
