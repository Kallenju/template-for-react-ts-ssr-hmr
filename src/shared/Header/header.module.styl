.example
  color #6195ed
  font-family Impact
  font-weight normal
  width 100%
  text-align center
