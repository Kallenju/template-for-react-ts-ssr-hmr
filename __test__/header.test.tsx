/*
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Header } from '../src/shared/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should increment counter', async () => {
    render(<Header />);

    const button = screen.getByRole('button', { name: /clicked/i });
    expect(button).toHaveTextContent('Clicked 0 times');

    await userEvent.click(button);
    expect(button).toHaveTextContent('Clicked 1 times');
  });
});
