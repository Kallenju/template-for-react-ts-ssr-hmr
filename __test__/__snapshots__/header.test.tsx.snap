// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <h1
    className="example"
  >
    Hello React
  </h1>
  <button
    className="example"
    onClick={[Function]}
  >
    Clicked 
    0
     times
  </button>
  <picture>
    <source
      height="118"
      srcSet="/static/images/raster/test-image/test-image.webp"
      type="image/webp"
      width="637"
    />
    <img
      alt="Test image"
      className="preview__image"
      height="118"
      src="/static/images/raster/test-image/test-image.jpg"
      width="637"
    />
  </picture>
</header>
`;
